workItems/workItemBlock.go

```diff
@@ -7,82 +7,69 @@ import (
 	"github.com/ElrondNetwork/elrond-go/core/check"
 	"github.com/ElrondNetwork/elrond-go/data"
 	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/indexer"
 	"github.com/ElrondNetwork/elrond-go/marshal"
 )
 
 var log = logger.GetOrCreate("core/indexer/workItems")
 
 type itemBlock struct {
-	indexer                saveBlockIndexer
-	marshalizer            marshal.Marshalizer
-	bodyHandler            data.BodyHandler
-	headerHandler          data.HeaderHandler
-	txPool                 map[string]data.TransactionHandler
-	signersIndexes         []uint64
-	notarizedHeadersHashes []string
-	headerHash             []byte
+	indexer       saveBlockIndexer
+	marshalizer   marshal.Marshalizer
+	argsSaveBlock *indexer.ArgsSaveBlockData
 }
 
 // NewItemBlock will create a new instance of ItemBlock
 func NewItemBlock(
 	indexer saveBlockIndexer,
 	marshalizer marshal.Marshalizer,
-	bodyHandler data.BodyHandler,
-	headerHandler data.HeaderHandler,
-	txPool map[string]data.TransactionHandler,
-	signersIndexes []uint64,
-	notarizedHeadersHashes []string,
-	headerHash []byte,
+	args *indexer.ArgsSaveBlockData,
 ) WorkItemHandler {
 	return &itemBlock{
-		indexer:                indexer,
-		bodyHandler:            bodyHandler,
-		headerHandler:          headerHandler,
-		txPool:                 txPool,
-		signersIndexes:         signersIndexes,
-		notarizedHeadersHashes: notarizedHeadersHashes,
-		marshalizer:            marshalizer,
-		headerHash:             headerHash,
+		indexer:       indexer,
+		marshalizer:   marshalizer,
+		argsSaveBlock: args,
 	}
 }
 
 // Save will prepare and save a block item in elasticsearch database
 func (wib *itemBlock) Save() error {
-	if check.IfNil(wib.headerHandler) {
+	if check.IfNil(wib.argsSaveBlock.Header) {
 		log.Warn("nil header provided when trying to index block, will skip")
 		return nil
 	}
 
-	log.Debug("indexer: starting indexing block", "hash", wib.headerHash, "nonce", wib.headerHandler.GetNonce())
+	log.Debug("indexer: starting indexing block",
+		"hash", logger.DisplayByteSlice(wib.argsSaveBlock.HeaderHash),
```

Comment:
why do you need `logger.DisplayByteSlice`? I think the hash was displayed correctly without it

Reply:
removed.